### PR TITLE
Improve signers handling in test framework

### LINKED ISSUE #2297: Improve test framework usability
It would be nice to add some public functionality into the test framework:
+ [NewSingle](https://github.com/nspcc-dev/neo-go/blob/v0.98.0/pkg/neotest/chain/chain.go#L114) and [NewMulti](https://github.com/nspcc-dev/neo-go/blob/v0.98.0/pkg/neotest/chain/chain.go#L135) returns committee multi-signer and it is impossible to invoke something that is signed by one of the members of the committee (not with its multi-signature). So now testing such functionality requires copy-pasting one of the [WIFs](https://github.com/nspcc-dev/neo-go/blob/v0.98.0/pkg/neotest/chain/chain.go#L20-L33) and constructing its signer.
+ `Signer` -> `PublicKey` conversion functionality. Now if a tester wants to get a public key, he has to use something like
```
node := contractInvoker.NewAccount(t)
nodePub, ok := vm.ParseSignatureContract(node.Script())
require.True(t, ok)
```

### PR DESCRIPTION
**Pull request by fyrchik**
Close #2297.

### CODE REVIEW

**Review by carpawell — APPROVED** [review]
From my side provided API is something that I expected to see. Not sure if failed tests are ok.

File: pkg/neotest/signer.go [hunk 1]
```diff
@@ -27,6 +27,14 @@ type Signer interface {
 	SignTx(netmode.Magic, *transaction.Transaction) error
 }
 
+// Signer is a generic interface which can be either simple-signature signer.
```
Comment (AnnaShaleva) [inline pkg/neotest/signer.go:31]: Remove `either`?

File: pkg/neotest/signer.go [hunk 2]
```diff
@@ -63,6 +71,11 @@ func (s *signer) SignTx(magic netmode.Magic, tx *transaction.Transaction) error
 	return (*wallet.Account)(s).SignTx(magic, tx)
 }
 
+// Account implements SingleSigner interface.
+func (s *signer) Account() *wallet.Account {
```
Comment (AnnaShaleva) [inline pkg/neotest/signer.go:85]: Do we really need this wrapper? I performed this cast manually in the previous tests.

Comment (AnnaShaleva) [reply pkg/neotest/signer.go:85]: Although it seems nice.

File: pkg/neotest/signer.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/neotest/signer.go:31]: And it's a `SingleSigner`

File: pkg/neotest/signer.go [hunk 3]
```diff
@@ -27,6 +27,14 @@ type Signer interface {
 	SignTx(netmode.Magic, *transaction.Transaction) error
 }
 
+// Signer is a generic interface which can be either simple-signature signer.
+type SingleSigner interface {
+	Signer
+	// Account returns underlying account which can be used to
+	// get public key and/or sign arbitrary things.
+	Account() *wallet.Account
```
Comment (roman-khimov) [inline pkg/neotest/signer.go:36]: I'm not sure we need this additional interface though, maybe `Signer` can have `Accounts() []*wallet.Account`?

File: pkg/neotest/signer.go [hunk 4]
```diff
@@ -35,11 +36,20 @@ type SingleSigner interface {
 	Account() *wallet.Account
 }
 
+type MultiSigner interface {
+	Signer
+	// Single returns simple-signature signer for n-th account in list.
+	Single(n int) SingleSigner
```
Comment (roman-khimov) [inline pkg/neotest/signer.go:42]: Unless we need to use `SingleSigner` received from this method as `Signer`, that'd be easier to do with the current interface scheme.

File: pkg/neotest/signer.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/neotest/signer.go:31]: I'll fix it to unblock #2299.

**Review by roman-khimov — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**